app/src/main/java/com/peluso/walletguru/ui/AddAccountFragment.kt

```diff
@@ -1,48 +1,141 @@
 package com.peluso.walletguru.ui
 
+import android.app.AlertDialog
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.ArrayAdapter
-import android.widget.ListView
+import android.widget.*
 import androidx.fragment.app.Fragment
 import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProvider
 import com.google.android.material.floatingactionbutton.FloatingActionButton
 import com.peluso.walletguru.R
+import com.peluso.walletguru.model.Account
+import com.peluso.walletguru.model.AccountType
 import com.peluso.walletguru.viewmodel.MainViewModel
 import com.peluso.walletguru.viewstate.MainViewState
+import java.lang.Float.parseFloat
 
 class AddAccountFragment : Fragment() {
 
     private lateinit var viewModel: MainViewModel
     private lateinit var listView: ListView
+    private lateinit var spinner: Spinner
+    private lateinit var newBalanceText: EditText
+    private lateinit var accountName: String
+    private lateinit var accountBalance: String
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
         val root = inflater.inflate(R.layout.fragment_addaccount, container, false)
+        accountName = "TBD"
+        accountBalance = "0"
         initViews(root)
-        viewModel =
-            ViewModelProvider(requireActivity()).get(MainViewModel::class.java)
+        viewModel = ViewModelProvider(requireActivity()).get(MainViewModel::class.java)
         return root
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        viewModel.viewState.observe(viewLifecycleOwner, Observer {
+            handleViewState(it)
+        })
+    }
+
     private fun initViews(view: View) {
         val syncButton: FloatingActionButton = view.findViewById(R.id.sync_accounts_button)
         listView = view.findViewById(R.id.accounts_listview)
         syncButton.setOnClickListener {
-            //TODO: make this pop up dialog to add account
+            launchSubmitNewAccount()
         }
     }
 
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        viewModel.viewState.observe(viewLifecycleOwner, Observer {
-            handleViewState(it)
-        })
+    private fun launchSubmitNewAccount() {
```

Comment:
this function should perform some sort of check which sees if there are any account types they are able to add. If I've added all the possible accounts, then this dialog should not open. Maybe then I see a toast message that says 'already added the maximum account types!'

Reply:
done